File: mobile/ios/Features/Shared/HostContextCard.swift
import SwiftUI

public struct HostContextCard: View {
    private enum AccessibilityID {
        static let card = "magichat.hostContext.card"
        static let title = "magichat.hostContext.title"
        static let detail = "magichat.hostContext.detail"
        static let endpoint = "magichat.hostContext.endpoint"
        static let device = "magichat.hostContext.device"
        static let activeInstance = "magichat.hostContext.activeInstance"
        static let offlineMessage = "magichat.hostContext.offlineMessage"
        static let refresh = "magichat.hostContext.refresh"
    }

    private let host: HostBeacon?
    private let presence: String?
    private let activeInstanceID: String?
    private let onRefreshStatus: (() -> Void)?
    private let refreshEnabled: Bool

    public init(
        host: HostBeacon?,
        presence: String?,
        activeInstanceID: String? = nil,
        onRefreshStatus: (() -> Void)? = nil,
        refreshEnabled: Bool = true
    ) {
        self.host = host
        self.presence = presence
        self.activeInstanceID = activeInstanceID
        self.onRefreshStatus = onRefreshStatus
        self.refreshEnabled = refreshEnabled
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            if let host {
                Text(host.displayName)
                    .font(.headline)
                    .accessibilityIdentifier(AccessibilityID.title)
                Text("\(host.transportLabel)\(presenceLabel)")
                    .font(.subheadline)
                    .foregroundStyle(.secondary)
                    .accessibilityIdentifier(AccessibilityID.detail)
                Text(host.endpointLabel)
                    .font(.footnote)
                    .foregroundStyle(.secondary)
                    .accessibilityIdentifier(AccessibilityID.endpoint)
                if let deviceID = host.deviceID, deviceID.isEmpty == false {
                    Text("Device: \(deviceID)")
                        .font(.footnote)
                        .foregroundStyle(.secondary)
                        .accessibilityIdentifier(AccessibilityID.device)
                }
                if let activeInstanceID, activeInstanceID.isEmpty == false {
                    Text("Active instance: \(activeInstanceID)")
                        .font(.footnote)
                        .foregroundStyle(.secondary)
                        .accessibilityIdentifier(AccessibilityID.activeInstance)
                }
                if host.canRunCommands == false {
                    Text("This host is offline right now. You can still manage pairings, but Team App commands are paused until it reconnects.")
                        .font(.footnote)
                        .foregroundStyle(.red)
                        .accessibilityIdentifier(AccessibilityID.offlineMessage)
                }
                if let onRefreshStatus {
                    Button("Check Host") {
                        onRefreshStatus()
                    }
                    .buttonStyle(.bordered)
                    .disabled(refreshEnabled == false)
                    .accessibilityIdentifier(AccessibilityID.refresh)
                }
            } else {
                Text("No host selected")
                    .font(.headline)
                    .accessibilityIdentifier(AccessibilityID.title)
                Text("Pair with or select a Team App host before launching, restoring, or sending prompts.")
                    .font(.footnote)
                    .foregroundStyle(.secondary)
                    .accessibilityIdentifier(AccessibilityID.detail)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(12)
        .background(.thinMaterial, in: RoundedRectangle(cornerRadius: 12, style: .continuous))
        .accessibilityIdentifier(AccessibilityID.card)
    }

    private var presenceLabel: String {
        let label = host?.presenceDisplayLabel ?? presence?.replacingOccurrences(of: "_", with: " ")
        guard let label, label.isEmpty == false else { return "" }
        return " • \(label)"
    }
}

File: mobile/android/app/src/main/java/com/magichat/mobile/ui/components/HostContextCard.kt
package com.magichat.mobile.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.Card
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.magichat.mobile.model.PairedHostRecord
import com.magichat.mobile.model.canRunCommands
import com.magichat.mobile.model.connectionModeLabel
import com.magichat.mobile.model.endpointLabel
import com.magichat.mobile.model.presenceLabel

@Composable
fun HostContextCard(
    host: PairedHostRecord?,
    presence: String?,
    activeInstanceId: String? = null,
    onRefreshStatus: (() -> Unit)? = null,
    refreshEnabled: Boolean = true,
    modifier: Modifier = Modifier,
) {
    Card(modifier = modifier.fillMaxWidth()) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            if (host == null) {
                Text("No host selected", style = MaterialTheme.typography.titleMedium)
                Text(
                    "Choose a saved host or pair a new one before launching or controlling Team App.",
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
                return@Column
            }

            val canRunCommands = host.canRunCommands(presence)
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
            ) {
                Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                    Text(host.displayName, style = MaterialTheme.typography.titleMedium)
                    Text(
                        host.endpointLabel(),
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                    )
                }
                if (onRefreshStatus != null) {
                    Button(
                        onClick = onRefreshStatus,
                        enabled = refreshEnabled,
                    ) {
                        Text("Check")
                    }
                }
            }

            Row(
                modifier = Modifier.horizontalScroll(rememberScrollState()),
                horizontalArrangement = Arrangement.spacedBy(8.dp),
            ) {
                StatusChip(
                    label = host.connectionModeLabel(),
                    background = MaterialTheme.colorScheme.secondaryContainer,
                    contentColor = MaterialTheme.colorScheme.onSecondaryContainer,
                )
                StatusChip(
                    label = host.presenceLabel(presence) ?: "unknown",
                    background = if (canRunCommands) Color(0xFF1E5E37) else MaterialTheme.colorScheme.errorContainer,
                    contentColor = if (canRunCommands) Color(0xFFD6F8DF) else MaterialTheme.colorScheme.onErrorContainer,
                )
                activeInstanceId?.takeIf { it.isNotBlank() }?.let {
                    StatusChip(
                        label = "Active $it",
                        background = MaterialTheme.colorScheme.primaryContainer,
                        contentColor = MaterialTheme.colorScheme.onPrimaryContainer,
                    )
                }
            }

            host.deviceId?.takeIf { it.isNotBlank() }?.let {
                Text(
                    "Device: $it",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }

            if (!canRunCommands) {
                Text(
                    "This host is offline or unreachable. You can still browse saved hosts, but session actions stay disabled until it reconnects.",
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.error,
                )
            }
        }
    }
}

@Composable
fun StatusChip(
    label: String,
    background: Color,
    contentColor: Color,
    modifier: Modifier = Modifier,
) {
    Text(
        text = label,
        style = MaterialTheme.typography.labelMedium,
        color = contentColor,
        modifier = modifier
            .background(background, RoundedCornerShape(999.dp))
            .padding(horizontal = 10.dp, vertical = 6.dp),
    )
}
